### PR TITLE
Create veBAL balances view on Dune V2 🎉

### PR DESCRIPTION
**Pull request by markusbkoch**
Port of V1's [view_vebal_balances](https://github.com/balancer-labs/dune-spellbook/blob/main/deprecated-dune-v1-abstractions/ethereum/balancer/view_vebal_balances.sql) to V2

*For Dune Engine V2*
I've checked that:

* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`

When you are ready for a review, tag duneanalytics/data-experience. We will re-open your forked pull request as an internal pull request. Then your spells will run in dbt and the logs will be avaiable in Github Actions DBT Slim CI. This job will only run the models and tests changed by your PR compared to the production project.

### CODE REVIEW

**Comment by jeff-dude** [comment]
thank you for the first spell PR!

while i look closer, two minor things (i've also created new PR to update PR template to check for this in builds #1468):
1. edit the dbt project yaml file at root level to add new balancer/ethereum directories in both models and seeds section (hopefully intuitive based on existing entries, but happy to help if needed)
2. edit the alter table macro under macros directory to include models here (ensure the alter statement is correct in terms of view vs. table) and added to both the sets and run query sections

File: models/balancer/ethereum/balancer_ethereum_sources.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+version: 2
+
+sources: 
+  - name: balancer_ethereum
+    description: "Ethereum decoded tables related to Balancer contracts"
+    freshness:
+      warn_after: { count: 12, period: hour }
+      error_after: { count: 24, period: hour }
```
Comment (jeff-dude) [inline models/balancer/ethereum/balancer_ethereum_sources.yml:8]: a recent item tracked was not all decoded tables get new data within 24-hour time period. i didn't check myself yet, but wanted to call out here if you know off the top of your head that the data in the decoded table should truly be refreshed / loaded this often. the timeframe can be extended if needed

File: models/balancer/ethereum/balancer_ethereum_vebal_balances_day.sql [hunk 2]
```diff
@@ -0,0 +1,133 @@
+WITH base_locks AS (
```
Comment (jeff-dude) [inline models/balancer/ethereum/balancer_ethereum_vebal_balances_day.sql:1]: most, if not all, models contain a config() block at the top with at minimum an 'alias' property applied, which allows you to specify the view name. without the alias, it'll default to the file name:
balancer_ethereum.balancer_ethereum_vebal_balances_day

the schema name will come from the dbt project yaml file, mentioned in prior comment

i'd assume you'd want to alias to:
balancer_ethereum.vebal_balances_day

Comment (jeff-dude) [reply models/balancer/ethereum/balancer_ethereum_vebal_balances_day.sql:1]: example: https://github.com/duneanalytics/spellbook/blob/main/models/prices/prices_tokens.sql

you can leave the schema property out if you apply correctly in dbt project file

**Comment by markusbkoch** [comment]
> * edit the dbt project yaml file at root level to add new balancer/ethereum directories in both models and seeds section (hopefully intuitive based on existing entries, but happy to help if needed)

addressed in 3f76a2e

**Comment by markusbkoch** [comment]
Thanks for the comments @jeff-dude - I started working on it but got sidetracked. Converting to draft until I can come back to it.

**Comment by jeff-dude** [comment]
hi @markusbkoch -- i'm going to close this PR, as there have been a good amount of changes to spellbook and such in the last month. it may be easiest to start fresh when you are ready, then i'll be more than happy to help finalize it 🙌

**Pull request closed without merging**